core/src/main/kotlin/xtdb/api/YamlSerde.kt

```diff
@@ -78,6 +80,34 @@ object StringWithEnvVarSerde : KSerializer<String> {
     }
 }
 
+object StringMapWithEnvVarsSerde : KSerializer<Map<String, String>> {
+    @OptIn(kotlinx.serialization.ExperimentalSerializationApi::class, kotlinx.serialization.InternalSerializationApi::class)
+    override val descriptor: SerialDescriptor = buildSerialDescriptor("StringMapWithEnvVarsSerde", StructureKind.MAP)
+
+    override fun serialize(encoder: Encoder, value: Map<String, String>) {
+        encoder.encodeString(value.toString())
```

Comment:
I doubt this'll serialize in valid YAML? if it's not used, I'd throw UOE instead - make it really clear :slightly_smiling_face:

Reply:
Yeah, that's a fair point - I'll do that, I think they do the same in the `kaml` tests as well :)